node/src/source.rs

```diff
@@ -742,8 +742,8 @@ mod test {
     #[test]
     fn test_fetcher() -> anyhow::Result<()> {
         let rpc = BitcoinRpc::new(
-            "http://127.0.0.1:18332",
-            BitcoinRpcAuth::UserPass("test".to_string(), "test".to_string()),
+            "http://127.0.0.1:48332",
```

Comment:
Since you're updating this test might as well move it to integration real quick for this pr

You could move it to `tests/fetcher.rs` or similar something like this :

```rust
    #[test]
    fn test_fetcher() -> Result<()> {
        let spaced = SpaceD::new()?;
        let fetcher_rpc = BitcoinRpc::new(
            &spaced.bitcoind.rpc_url(),
            BitcoinRpcAuth::UserPass("user".to_string(), "password".to_string()),
        );
        let miner_addr = spaced.bitcoind.client.get_new_address(None, None)?.assume_checked();
        const GENERATED_BLOCKS: u32 = 10;
        spaced.bitcoind.client.generate_to_address(GENERATED_BLOCKS as u64, &miner_addr)?;

        let client = reqwest::blocking::Client::new();
        let (fetcher, receiver) = BlockFetcher::new(
            fetcher_rpc.clone(), client.clone(), 8,
        );
        fetcher.start(ChainAnchor {
            hash: fetcher_rpc.send_json_blocking(&client, &fetcher_rpc.get_block_hash(0))?,
            height: 0,
        });

        let mut start_block = 0;
        let timeout = Duration::from_secs(5);
        let start_time = Instant::now();

        loop {
            if start_time.elapsed() > timeout {
                panic!("Test timed out after {:?}", timeout);
            }
            match receiver.try_recv() {
                Ok(BlockEvent::Block(id, _)) => {
                    start_block += 1;
                    if id.height == GENERATED_BLOCKS {
                        break;
                    }
                }
                Ok(BlockEvent::Error(e)) => panic!("Unexpected error: {}", e),
                Err(TryRecvError::Empty) => {
                    std::thread::sleep(Duration::from_millis(10));
                }
                Err(TryRecvError::Disconnected) => panic!("Disconnected unexpectedly"),
            }
        }
        assert_eq!(start_block, GENERATED_BLOCKS, "Not all blocks were received");
        Ok(())
    }
```

Reply:
I thought we can do this later, but I've made this change now, thanks!